src/itsdangerous/encoding.py

```diff
@@ -1,26 +1,32 @@
 import base64
 import string
 import struct
+import typing as _t
 
 from .exc import BadData
 
+if _t.TYPE_CHECKING:
+    _t_str_bytes = _t.Union[str, bytes]
 
-def want_bytes(s, encoding="utf-8", errors="strict"):
+
+def want_bytes(
+    s: "_t_str_bytes", encoding: str = "utf-8", errors: str = "strict"
+) -> bytes:
     if isinstance(s, str):
         s = s.encode(encoding, errors)
 
     return s
 
 
-def base64_encode(string):
+def base64_encode(string: "_t_str_bytes") -> bytes:
```

Comment:
This is definitely a stylistic question, but I find this tricky to read and understand. The type hints for me are much better when explicit for documentation purposes. 

Also is `AnyStr` not appropriate for this?

Reply:
I got tired of writing the full syntax over and over again. This one is fairly simple to replace, but since I was making aliases anyway for more complex types, and this was used so much, I made it an alias too. If you think this is too distracting I can change it, although I'd want to know what the cutoff is for using an alias or not. Note that PyCharm and Mypy both report the real type, not the alias.

`AnyStr` is a `TypeVar`, so its intended use is for ensuring consistency across arguments/return. The `Union` alias is more "anything goes", which represents how ItsDangerous works right now. See my discussion on str and bytes.

Reply:
I feel like the types are an additional piece of documentation (they also show up to developers, mostly when using IDEs and sometimes in documentation) so it's nice if the names are meaningful.

Reply:
Ah, Sphinx does *not* dereference the aliases, so it does look weird there. Maybe there's a Sphinx plugin for this?